crates/primitives/src/block.rs

```diff
@@ -71,6 +76,17 @@ impl Block {
         BlockWithSenders { block: self, senders }
     }
 
+    /// **Expensive**. Transform into a [`BlockWithSenders`] by recovering senders in the contained
+    /// transactions.
+    ///
+    /// Returns `None` if the number of senders does not match the number of transactions in the
+    /// block and the signer recovery for one of the transactions fails.
+    #[track_caller]
+    pub fn with_recovered_senders(self) -> Option<BlockWithSenders> {
+        let senders = self.senders()?;
+        Some(BlockWithSenders { block: self, senders })
```

Comment:
should we return an `Err(Self)` here?

Reply:
we don't use that anywhere so doesn't make sense imo